lib/ansible/modules/crypto/get_certificate.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: get_certificate
+author: "John Westcott IV"
+version_added: "2.7"
+short_description: Get a certificate from a host:port
+description:
+    - Makes an SSL connection and returns information about the presented certificate
+options:
+    host:
+      description:
+        - The host to get the cert for (IP is fine)
+      required: True
+    ca_certs:
+      description:
+        - A file containing a list of root certificates; if present, the cert will be validated against these root certs.
+      required: False
+    port:
+      description:
+        - The port to connect to
+      required: True
+    timeout:
+      description:
+        - The timeout in seconds
+      required: False
+      default: 10
+
+notes:
+  - When using ca_certs on OS X it has been reported that in some conditions the validate will always suceed.
+
+requirements:
+  - "python >= 2.6"
+  - "ssl"
+  - "python-pyOpenSSL >= 0.15"
+  - "socket"
+'''
+
+RETURN = '''
+cert:
+    description: Information about the cert (CN, OU, etc)
```

Comment:
Line 163++ seems like this only contains information about the subject, not the certificate itself.

Reply:
That is correct. I've changed the wording so that cert is an alias to object. Cert was the original return from this module so I'd like to leave it in unless there are strong objections.

Reply:
I'd rather see/expect a string that contains the whole certificate (e.g. in PEM format) in this return value, not the subject.